src/decoder/lattice-faster-online-decoder.cc

```diff
@@ -891,7 +894,7 @@ BaseFloat LatticeFasterOnlineDecoder::ProcessEmitting(
     StateId state = best_elem->key;
     Token *tok = best_elem->val;
     cost_offset = - tok->tot_cost;
-    for (fst::ArcIterator<fst::Fst<Arc> > aiter(fst_, state);
+    for (fst::ArcIterator<FstType> aiter(fst, state);
          !aiter.Done();
          aiter.Next()) {
       Arc arc = aiter.Value();
```

Comment:
I just realized that there are some obvious optimizations here that can be done.
It seems likely that makeing Arc a const reference (i.e. `const Arc &arc...`) might be more efficient.

And (this is related), below, instead of `arc.weight == ...`, it would be more efficient to just work in cost space, i.e. do:
  `BaseFloat new_weight = arc.weight.Value() + cost_offset - decodable->LogLikelihood(frame, arc.ilabel);`
(keep within the 80-char limit).

Wherever you have these non-const references to Fst, make them const references.  And if there is anywhere else where there are unnecessary copies of Arc, then change that too.  This particular case isn't that important because it's only done for the best token, not all of them.